internal/services/serviceprincipals/synchronization_job_resource.go

```diff
@@ -0,0 +1,251 @@
+	client := meta.(*clients.Client).ServicePrincipals.SynchronizationJobClient
+	spClient := meta.(*clients.Client).ServicePrincipals.ServicePrincipalsClient
+	objectId := d.Get("service_principal_id").(string)
+
+	tf.LockByName(servicePrincipalResourceName, objectId)
+	defer tf.UnlockByName(servicePrincipalResourceName, objectId)
+
+	servicePrincipal, status, err := spClient.Get(ctx, objectId, odata.Query{})
+	if err != nil {
+		if status == http.StatusNotFound {
+			return tf.ErrorDiagPathF(nil, "service_principal_id", "Service principal with object ID %q was not found", objectId)
+		}
+		return tf.ErrorDiagPathF(err, "service_principal_id", "Retrieving service principal with object ID %q", objectId)
+	}
+	if servicePrincipal == nil || servicePrincipal.ID == nil {
+		return tf.ErrorDiagF(errors.New("nil service principal or service principal with nil ID was returned"), "API error retrieving service principal with object ID %q", objectId)
+	}
+
+	// Create a new synchronization job
+	synchronizationJob := msgraph.SynchronizationJob{
+		TemplateId: utils.String(d.Get("template_id").(string)),
+	}
+
+	newJob, _, err := client.Create(ctx, synchronizationJob, *servicePrincipal.ID)
+	if err != nil {
+		return tf.ErrorDiagF(err, "Creating synchronization job for service principal ID %q", *servicePrincipal.ID)
+	}
+	if newJob == nil {
+		return tf.ErrorDiagF(errors.New("nil received when creating synchronization Job"), "API error creating synchronization job for service principal ID %q", *servicePrincipal.ID)
+	}
+	if newJob.ID == nil {
+		return tf.ErrorDiagF(errors.New("nil or empty id received"), "API error creating synchronization job for service principal ID %q", *servicePrincipal.ID)
+	}
+
+	id := parse.NewSynchronizationJobID(*servicePrincipal.ID, *newJob.ID)
+
+	// Wait for the job to appear, this can take several moments
+	timeout, _ := ctx.Deadline()
+	_, err = (&resource.StateChangeConf{
+		Pending:                   []string{"Waiting"},
+		Target:                    []string{"Done"},
+		Timeout:                   time.Until(timeout),
+		MinTimeout:                1 * time.Second,
+		ContinuousTargetOccurence: 5,
+		Refresh: func() (interface{}, string, error) {
+			_, status, err := client.Get(ctx, id.JobId, id.ServicePrincipalId)
+			if err != nil {
+				if status == http.StatusNotFound {
+					return "stub", "Waiting", nil
+				}
+				return nil, "Error", fmt.Errorf("retrieving member")
+			}
+			return "stub", "Done", nil
+		},
+	}).WaitForStateContext(ctx)
+
+	if err != nil {
+		return tf.ErrorDiagF(err, "Waiting for synchronization job %q", id.JobId)
+	}
+
+	d.SetId(id.String())
+
+	// Start job if desired
+	desiredState := d.Get("state").(string)
+	if desiredState == "Active" {
+		_, err := client.Start(ctx, id.JobId, id.ServicePrincipalId)
+		if err != nil {
+			return tf.ErrorDiagF(err, "Starting synchronization job %q", id.JobId)
+		}
+	}
```

Comment:
Maybe something like this is we are to change this to a boolean?
```suggestion
	if d.Get("enabled").(bool) {
		_, err := client.Start(ctx, id.JobId, id.ServicePrincipalId)
		if err != nil {
			return tf.ErrorDiagF(err, "Starting synchronization job %q", id.JobId)
		}
	}
```